File: main/java/com/example/recipesandcookingtips/SplashActivity.java
package com.example.recipesandcookingtips;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
    }new handler ().postDelyed(new Runnable() {

        @Override
                public void run() {
            Intent intent = new Intent(SplashActivity.this,MainActivity.class);
            startActivities(intent);
            finish();

        },3000);
    }
}

File: src/main/java/com/example/recipesandcookingtips1/SplashActivity.kt
package com.example.recipesandcookingtips1

import android.content.Intent
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_splash)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        Thread(Runnable {
            Thread.sleep(1500)
            goToMainActivity()
        }).start()
    }
    fun goToMainActivity(){
        Thread.sleep(2000)
        startActivity(Intent(this,MainActivity::class.java).setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP))
        finish()
    }
}